apps/framework-cli-e2e/test/templates.test.ts

```diff
@@ -358,6 +358,84 @@ const utils = {
     }
   },
 
+  verifyVersionedConsumptionApi: async (
+    endpoint: string,
+    expectedResponse: any,
+  ): Promise<void> => {
+    const response = await fetch(
+      `${TEST_CONFIG.server.url}/consumption/${endpoint}`,
+    );
+    if (response.ok) {
+      console.log("Versioned API test request sent successfully");
+      const json = (await response.json()) as any[];
+
+      // Check structure and value validation rather than exact matches
+      expect(json).to.be.an("array");
+      expect(json.length).to.be.at.least(1);
+
+      json.forEach((item: any, index: number) => {
+        const expected = expectedResponse[index] || expectedResponse[0];
+
+        // Verify structure - all expected keys exist
+        Object.keys(expected).forEach((key) => {
+          if (key === "metadata") {
```

Comment:
This is hardcoded to what you expect - do you think there is a way to have this logic be generic? since you are passing what's expected as the input of the function